contracts/Remittance.sol

```diff
@@ -1,16 +1,118 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(address => uint) public balances;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
+
+    struct RemittanceInstance {
+        address funder;
+        address broker;
+        uint amount;
+        bool fundsReleased;
+    }
+
+    event Deposit(
+        address indexed depositor,
+        uint amount
+    );
+
+    event RemittanceCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint amount
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        address indexed broker,
+        uint amount
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    // Generic deposit function
+    function deposit() external payable whenNotPaused {
+        require(msg.value > 0, "The value must be greater than 0");
+        balances[msg.sender] = balances[msg.sender].add(msg.value);
+
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    // The funder can create a remittance
+    function create(bytes32 hash, address broker, uint256 amount)
```

Comment:
What could this mean if the `hash == 0`?

Reply:
> What could this mean if the `hash == 0`?

Interesting. I initially though that a valid bytes32 could not be sent with 0 but tried:

`const hash = web3.utils.fromAscii(0)`

The contract accepted it!

I have added in:

`require(hash.length > 0, "Hash cannot be empty");`

Reply:
If `hash` is a `bytes32`, then its `.length` is always 32. What I was asking is what happens if all the bits of `hash` are `0`? I.e. `hash == 0x000...000`.